cipher/src/block.rs

```diff
@@ -386,6 +417,17 @@ pub trait BlockEncryptMut: BlockSizeUser + Sized {
         let buf = InOutBufReserved::from_slices(msg, out_buf).map_err(|_| PadError)?;
         self.encrypt_padded_inout_mut::<P>(buf)
     }
+
+    /// Pad input and encrypt into a newly allocated Vec. Returns resulting ciphertext Vec.
+    #[cfg(all(feature = "block-padding", feature = "alloc"))]
+    #[cfg_attr(docsrs, doc(cfg(all(feature = "block-padding", feature = "alloc"))))]
+    #[inline]
+    fn encrypt_padded_vec_mut<P: Padding<Self::BlockSize>>(self, msg: &[u8]) -> Vec<u8> {
+        let mut out = allocate_out_vec::<Self>(msg.len());
+        self.encrypt_padded_b2b_mut::<P>(msg, &mut out)
+            .expect("enough space for encrypting is allocated");
```

Comment:
It would be nice to check whether the compiler is smart enough to eliminate this `expect`. I can do it later myself.